packages/next-auth/src/providers/saml-jackson.ts

```diff
@@ -0,0 +1,37 @@
+import type { OAuthConfig, OAuthUserConfig } from "."
+
+export interface SAMLJacksonProfile {
+  id: string;
+  email: string;
+  firstName: string | null | undefined;
+  lastName: string | null | undefined;
+}
+
+export default function SAMLJackson<
+  P extends Record<string, any> = SAMLJacksonProfile
+>(options: OAuthUserConfig<P>): OAuthConfig<P> {
+  return {
+    id: "saml-jackson",
+    name: "BoxyHQ SAML Jackson",
+    type: "oauth",
+    version: "2.0",
+    checks: ["pkce", "state"],
+    authorization: {
+      url: `${options.issuer}/api/oauth/authorize`,
+      params: {
+        provider: "saml",
+      },
+    },
+    token: `${options.issuer}/api/oauth/token`,
+    userinfo: `${options.issuer}/api/oauth/userinfo`,
+    profile(profile) {
+      return {
+        id: profile.id,
+        email: profile.email || "",
+        name: [profile.firstName, profile.lastName].filter(Boolean).join(" "),
+        image: null,
```

Comment:
```suggestion
        email: profile.email || "",
        name: [profile.firstName, profile.lastName].filter(Boolean).join(" "),
        image: null,
```

In which case will the user not receive an email here?

Is there any API to get a profile picture?

Reply:
Unfortunately SAML does not provide a way to get a profile picture